gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-client/src/test/java/org/apache/gobblin/service/FlowConfigV2Test.java

```diff
@@ -301,6 +302,42 @@ public void testLocalGroupOwnershipUpdates() throws Exception {
     _client.deleteFlowConfig(new FlowId().setFlowGroup(TEST_GROUP_NAME).setFlowName(TEST_FLOW_NAME_7));
   }
 
+
+  @Test (expectedExceptions = RestLiResponseException.class)
```

Comment:
I think it would be a better practice for us to try and catch the exception, and then assert that the error code is as expected. For example, with `testGroupRequesterRejected`
```
try {
...
} catch (RestLiResponseException e) {
  Assert.assertEquals(e.getStatus(), HttpStatus.ORDINAL_401_Unauthorized);
}
```

I was able to catch a bug before in this test suite as it actually returned a different error code than expected. Also these tests would mark 500 errors as passing tests which I think we should avoid.
Also it allows better granularity in the test too, so we can assert that only the second call here would fail

Reply:
Doesn't this have the issue that the test will pass if no exception is thrown? (Unless you do something like `throw new RuntimeException("Shouldn't reach here")` after the point you expect it to fail). Probably better to use the `expectedExceptionsMessageRegExp` mentioned by Alex above.

Reply:
Try/catch will also work too. I usually add Assert.fail("Expected exception X") at the end of the try block.

Junit has this exception verification streamlined with assertThrows, but we're still on TestNg - https://stackoverflow.com/questions/40268446/junit-5-how-to-assert-an-exception-is-thrown

Reply:
Actually I think I prefer it that way, because we can just check the HTTP status code instead, checking error messages seems a bit worse because if we ever want to change the message sent it will affect all the tests.